src/font/Metrics.zig

```diff
@@ -188,6 +192,19 @@ pub fn calc(face: FaceMetrics) Metrics {
     // We do cap it at `cell_height` though for obvious reasons.
     const icon_height = @min(cell_height, cap_height * 1.2);
 
+    // We set icon_width using the icon_height / cell_height
+    // ratio, such that
+    // icon_width / (2 * cell_width) == icon_height / cell_height
+    // This only applies when the constraint width is two cells;
+    // icons constrained to a single cell can take up the full
+    // cell width.
+    //
+    // It's important that the icon_width is calculated here using
+    // the cell_width and cell_height derived from the face, and
+    // not in Constraint.constrain, which is called after the
+    // adjust-cell-{width,height} modifiers have been applied.
+    const icon_width = 2 * cell_width * icon_height / cell_height;
```

Comment:
The PR as-is preserves the heuristic of using the same margin-to-content ratio for the width and height. However, I've been playing a lot with this today, and I think setting `icon_width = c * cell_width` for some fixed `c` works even better across different primary fonts with different cell aspect ratios. My suggestion would be `c = 4.0 / 3.0` (slightly larger could also work, up to about 1.5).

This decoupling would also let us increase the height a bit for tall icons. Incidentally, I'd also suggest `icon_height = (4.0 / 3.0) * cap_height`.

LMK if I should push these changes, or if you want to discuss over some screenshots on Discord.